document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -1093,6 +1098,25 @@ protected Document prepareDocument() throws SQLException, IOException {
       return new JSONDocument(MAPPER.writeValueAsString(jsonNode));
     }
 
+    private String getColumnValue(
+        ResultSetMetaData resultSetMetaData, String columnName, int columnIndex)
+        throws SQLException, JsonProcessingException {
+      int columnType = resultSetMetaData.getColumnType(columnIndex);
+      // check for array
+      if (columnType == Types.ARRAY) {
+        return MAPPER.writeValueAsString(resultSet.getArray(columnIndex).getArray());
+      }
+
+      // check for ID column
+      if (PostgresUtils.OUTER_COLUMNS.contains(columnName)
+          && columnName.equals(PostgresUtils.ID_COLUMN)) {
```

Comment:
Doesn't the second condition (ID_COLUMN) directly imply the first condition (OUTER_COLUMNS).?

Reply:
Yah, right. Changing it.

Reply:
Done.